compiler/haskell-ide-core/src/Development/IDE/Functions/Compile.hs

```diff
@@ -233,7 +233,12 @@ runGhcSession IdeOptions{..} modu env act = runGhcEnv env $ do
 moduleImportPaths :: GHC.ParsedModule -> Maybe FilePath
 moduleImportPaths pm
   | rootModDir == "." = Just rootPathDir
-  | otherwise = dropTrailingPathSeparator <$> stripSuffix rootModDir rootPathDir
+  | otherwise =
+    -- TODO (MK) stripSuffix (normalise rootModDir) (normalise rootPathDir)
+    -- would be a better choice but at the moment we do not consistently
+    -- normalize file paths in the Shake graph so we can end up with the
+    -- same module being represented twice in the Shake graph.
+    Just $ dropTrailingPathSeparator $ dropEnd (length rootModDir) rootPathDir
```

Comment:
What is rootPathDir doesn't start with rootModDir? I guess the alternative is adding nothing to the import path, so adding something wrong probably won't be harmful...

Reply:
Yeah I don’t think it should cause issues. The whole area around file paths and include dirs deserves some refactoring and cleanup once the LSP side is done so I’ll leave it for that.